src/Word.jl

```diff
@@ -0,0 +1,68 @@
+abstract type Word{T} <: AbstractVector{T} end
```

Comment:
In general one would write a (markdown formatted) docstring above the definition
```julia
"""
    Word{T}
Short description

Longer description
The subtypes of `Word{T}` need to implement the following methods
 * `Base.one` the monoid identity element (empty word)
 * `Base.:*` for monoid binary operation (concatenation)
 * iteration protocol
 * `length` the length of word in the appropriate word-length metric
....
"""
abstract type Word{T} <:AbstractVector{T}
```

this way when you type `?Word` in REPL You will see the above docstring